plugin/trino-hive/src/main/java/io/trino/plugin/hive/CachingDirectoryLister.java

```diff
@@ -91,10 +102,36 @@ private static SchemaTablePrefix parseTableName(String tableName)
         if (tablePrefixes.stream().noneMatch(prefix -> prefix.matches(table.getSchemaTableName()))) {
             return iterator;
         }
-        return cachingRemoteIterator(iterator, path);
+        return cachingRemoteIterator(iterator, table, path);
+    }
+
+    @Override
+    public void schemaTableUpdated(SchemaTableName schemaTableName)
+    {
+        if (tablePrefixes.isEmpty()) {
+            // Caching functionality is not being used
+            return;
+        }
+        if (tablePrefixes.stream().anyMatch(prefix -> prefix.matches(schemaTableName))) {
+            // In the absence of a cache with a hierarchical structure, all the keys
+            // from the cache need to be checked to see whether they belong to the table path which
+            // is to be invalidated.
+            List<TablePath> affectedPaths = new ArrayList<>();
+            // As per guava cache javadoc: the iterators from the map returned by the method `cache.asMap()`
+            // are at least <i>weakly consistent</i> and they are safe for concurrent use.
```

Comment:
We don't have a convention of copying important pieces of documentation into the source code.
We hope the assumptions won't change.